roles/sap_ha_install_anydb_ibmdb2/tasks/main.yml

```diff
@@ -0,0 +1,165 @@
+      community.general.zypper:
+        name: "{{ __sap_ha_install_anydb_ibmdb2_pcmk_rpm_files.files | map(attribute='path') | list }}"
+        state: present
+        disable_gpg_check: true
+      when: ansible_os_family == 'Suse'
+
+    # SAP Note 3100287 - DB6: Db2 Support for Pacemaker Cluster Software
+    - name: SAP HA AnyDB - IBM Db2 HADR - Verify IBM Db2 'Integrated Linux Pacemaker' is installed
+      register: __sap_ha_install_anydb_ibmdb2_pcmk_version_check
+      ansible.builtin.shell: |
+        rpm -q corosync | grep -I db2pcmk
+        rpm -q pacemaker | grep -I db2pcmk
+        rpm -q crmsh | grep -I db2pcmk
+
+    - name: SAP HA AnyDB - IBM Db2 HADR - Display IBM Db2 'Integrated Linux Pacemaker' installation details
+      ansible.builtin.debug:
+        msg: "{{ __sap_ha_install_anydb_ibmdb2_pcmk_version_check.stdout_lines }}"
+
+    - name: SAP HA AnyDB - IBM Db2 HADR - Identify IBM Db2 cluster manager (db2cm) utility
+      ansible.builtin.find:
+        paths: /db2
+        recurse: true
+        file_type: file
+        patterns: db2cm
+      register: __sap_anydb_ibmdb2_db2cm
+
+    - name: SAP HA AnyDB - IBM Db2 HADR - Ensure directory for db2cm binary
+      ansible.builtin.file:
+        path: /db2/db2{{ sap_ha_install_anydb_ibmdb2_sid | lower }}/sqllib/bin
+        mode: "0755"
+        state: directory
+
+    - name: SAP HA AnyDB - IBM Db2 HADR - Copy db2cm binary to binary path
+      ansible.builtin.copy:
+        src: "{{ __sap_anydb_ibmdb2_db2cm.files[0].path }}"
+        remote_src: true
+        dest: /db2/db2{{ sap_ha_install_anydb_ibmdb2_sid | lower }}/sqllib/bin
+        mode: '0755'
+
+    # - name: SAP HA AnyDB - IBM Db2 HADR - Verify /usr/lib/ocf/resource.d/heartbeat contains Db2agents (db2hadr, db2inst, db2ethmon)
+
+    - name: SAP HA AnyDB - IBM Db2 HADR - Execute db2installPCMK shell script
+      ansible.builtin.shell: |
+        {{ __sap_ha_install_anydb_ibmdb2_pcmk.files[0].path }} -i
+
+
+- name: SAP HA AnyDB - IBM Db2 HADR - Ansible Task Block to create Linux Pacemaker cluster from AnyDB Primary node
+  when: sap_ha_install_anydb_ibmdb2_hostname_primary == inventory_hostname_short
+  block:
+
+    # db2cm script must run as root
+    - name: SAP HA AnyDB - IBM Db2 HADR - Create Linux Pacemaker cluster using db2cm
+      register: __sap_ha_install_anydb_ibmdb2_config
+      ansible.builtin.shell: |
+        export ibmdb2_sid="{{ sap_ha_install_anydb_ibmdb2_sid }}"
+        export ibmdb2_instance="db2{{ sap_ha_install_anydb_ibmdb2_sid | lower }}"
+        export anydb_primary="{{ sap_ha_install_anydb_ibmdb2_hostname_primary }}"
+        export anydb_secondary="{{ sap_ha_install_anydb_ibmdb2_hostname_secondary }}"
+        # Assume 1 OS Network Interface until improvements can be made
+        export anydb_primary_nic="{{ hostvars[sap_ha_install_anydb_ibmdb2_hostname_primary].ansible_default_ipv4.interface }}"
```

Comment:
Looks like role needs facts to be gathered, this is not mentioned in the role documentation and facts are not collected as part of the role. Please clarify and document what facts are required for the role.

Reply:
Why? `ansible_facts` are gathered automatically

Reply:
Added below just in case:
```yaml
- name: SAP HA AnyDB - IBM Db2 HADR - Collect required facts
  ansible.builtin.setup:
    gather_subset: hardware,interfaces
```